src/modules/72-templates-library/components/TemplateStudio/PipelineTemplateCanvas/PipelineTemplateCanvas.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { useParams } from 'react-router-dom'
+import { omit, set } from 'lodash-es'
+import { TemplateContext } from '@templates-library/components/TemplateStudio/TemplateContext/TemplateContext'
+import type { GitQueryParams, ProjectPathProps } from '@common/interfaces/RouteInterfaces'
+import { useGlobalEventListener, useQueryParams } from '@common/hooks'
+import { TemplatePipelineProvider } from '@pipeline/components/TemplatePipelineContext'
+import { sanitize } from '@common/utils/JSONUtils'
+import type { PipelineInfoConfig } from 'services/cd-ng'
+import { PipelineContextType } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineContext'
+import { DrawerTypes } from '@templates-library/components/TemplateStudio/TemplateContext/TemplateActions'
+import { RightDrawer } from '@templates-library/components/TemplateStudio/RightDrawer/RightDrawer'
+import StageBuilder from '@pipeline/components/PipelineStudio/StageBuilder/StageBuilder'
+import { RightBar as PipelineStudioRightBar } from '@pipeline/components/PipelineStudio/RightBar/RightBar'
+import { TemplateDrawer } from '@templates-library/components/TemplateDrawer/TemplateDrawer'
+
+export const DefaultNewPipelineName = 'Pipeline Name'
+export const DefaultNewPipelineId = 'pipeline_name'
+
+const PipelineTemplateCanvas = () => {
+  const {
+    state: {
+      templateView: { isDrawerOpened }
+    }
+  } = React.useContext(TemplateContext)
+  const {
+    state: { template, isLoading, isUpdated, templateView },
+    updateTemplate,
+    updateTemplateView,
+    isReadonly
+  } = React.useContext(TemplateContext)
+  const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
+  const { branch, repoIdentifier } = useQueryParams<GitQueryParams>()
+
+  const createPipelineFromTemplate = () => ({
+    ...template.spec,
+    name: DefaultNewPipelineName,
+    identifier: DefaultNewPipelineId
+  })
+
+  const [pipeline, setPipeline] = React.useState<PipelineInfoConfig>(createPipelineFromTemplate())
+
+  React.useEffect(() => {
+    if (!isLoading && !isUpdated) {
+      setPipeline(createPipelineFromTemplate())
+    }
+  }, [isLoading, isUpdated])
```

Comment:
Codacy has a fix for the issue: [React Hook React.useEffect has a missing dependency: 'createPipelineFromTemplate'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232314)
```suggestion
  }, [createPipelineFromTemplate, isLoading, isUpdated])
```